src/Message/ItemDataTrait.php

```diff
@@ -9,15 +9,15 @@
 trait ItemDataTrait
 {
     /**
-     * @param ItemBag $items
+     * @param ItemBag|null $items
      *
      * @return array[]
      */
-    public function getItemData(ItemBag $items): array
+    public function getItemData(ItemBag $items = null): array
```

Comment:
I would rather us an empty array as default and avoid the null sitation altogether.

What is the usecase, when exactly is null provided as param?

Reply:
> What is the usecase, when exactly is null provided as param?

> This is an issue when trying to remove all items from a cart, and keeping the transaction in sync.

Reply:
I can change the `AbstractOrderTrait` instead with `'order_lines' => $this->getItemData($this->getItems() ?? new ItemBag()),`

Reply:
best solution ever! :+1: